packages/gatsby/src/services/graphql-typegen.ts

```diff
@@ -28,12 +28,18 @@ export async function graphQLTypegen({
   )
   activity.start()
 
-  const { schema, definitions } = store.getState()
+  const { schema, definitions, config } = store.getState()
+  const graphqlTypegenOptions = config.graphqlTypegen as IGraphQLTypegenOptions
```

Comment:
Second part of https://github.com/gatsbyjs/gatsby/pull/35966/files#r902618646 - having to type cast here is the smell for me - we shouldn't need to typecast things in our redux store anymore as they should be processed and normalized already

Reply:
The problem with our redux type definitions (not specific to graphqlTypegen) is that we use `?` everywhere:

```
graphqlTypegen?: IGraphQLTypegenOptions
```

So this will be not the type I want/need it:

![image](https://user-images.githubusercontent.com/16143594/174812742-9fe7ecea-b71f-4548-930f-17bb5a124767.png)

Because at this stage it can't be undefined